core/silkworm/chain/config.hpp

```diff
@@ -72,6 +73,9 @@ struct ChainConfig {
     // https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/arrow-glacier.md
     std::optional<uint64_t> arrow_glacier_block{std::nullopt};
 
+    // PoW to PoS switch; see EIP-3675
+    std::optional<intx::uint256> terminal_total_difficulty{std::nullopt};
```

Comment:
Maybe a matter of preference, but the initializer `{std::nullopt}` is not needed, as by default `std::optional` is constructed empty (what else could it contain?).